tools/c7n_gcp/c7n_gcp/policy.py

```diff
@@ -101,8 +103,47 @@ def run(self, event, context):
         return PullMode.run(self)
 
 
+class FunctionModeImpl(FunctionMode):
+    def resolve_resources(self, event):
+        raise NotImplementedError("subclass responsibility")  # pragma: no cover
+
+    def run(self, event, context):
```

Comment:
If this is the behavior for all `FunctionMode` subclasses, might it be clearer to move this implementation to there and avoid the extra layer of abstraction?

Reply:
I had added a layer because PeriodicMode also used FunctionMode, but since it never calls resolver_resources, we're probably fine.